UpdatedSyntheticDataset/SyntheticDataset2/synthetic_dataset_maker.py

```diff
@@ -49,15 +49,15 @@ def create_target_maps(number_of_target_maps, number_of_targets_on_each_map):
         print("====================================")
 
     @staticmethod
-    def run_single_targets_generator(number_of_single_targets, process_number):
+    def run_single_targets_generator(number_of_single_targets, process_number, shape_type, shape_orientation, letter, shape_color, letter_color, rotation):
         for index in range(number_of_single_targets):
-            single_target = RandomTargetWithBackground()
+            single_target = RandomTargetWithBackground(shape_type, shape_orientation, letter, shape_color, letter_color, rotation)
             single_target.create_random_target_with_background()
             Logger.log("Current process: " + str((process_number / number_of_single_targets) + 1) + "\n")
             single_target.record_random_target_with_background(process_number + index + 1)
 
     @staticmethod
-    def create_single_targets(number_of_single_targets):
+    def create_single_targets(number_of_single_targets, shape_type, shape_orientation, letter, shape_color, letter_color, rotation):
```

Comment:
You should not make the user choose this as the only option for creating a dataset. You should have two methods here, one to create a random result, and one where the user can specify it.